webpack/stories/docs/APIMiddlewareUsage.md

```diff
@@ -0,0 +1,150 @@
+# Example: how to use the API middleware
+
+The example API returns a JSON object like this:
+```json
+{
+  "items": [
+    {"id": 319, "title": "setting", "action": "update"},
+    {"id": 150, "title": "bookmark", "action": "create"}
+  ],
+}
+```
+
+```js
+  /** MyComponent.js*/
+  import React from 'react';
+  import { STATUS } from '../../constants';
+
+  const MyComponent = ({ status, error, items }) => {
+    if (status === STATUS.PENDING ) {
+      return <div>Loading...</div>
+    }
+
+    if (status === STATUS.ERROR) {
+      return <div>Error: {error.message}</div>
+    }
+
+    return (
+      <ul>
+        {items.map(item => (
+          <li key={item.id}>
+            {item.title} {item.action}
+          </li>
+        ))}
+      </ul>
+    );
+  }
+```
+
+```js
+  /** index.js*/
+  import React, { useEffect } from 'react';
+  import PropTypes from 'prop-types';
+  import { useSelector, useDispatch } from 'react-redux';
+  import { selectItems, selectStatus, selectError } from './MyComponentsSelectors.js';
+  import { getData } from './MyComponentActions';
+  import MyComponent from './MyComponent';
+
+  const ConnectedMyComponent = ({ path }) => {
+    const items = useSelector(selectItems);
+    const status = useSelector(selectStatus);
+    const error = useSelector(selectError);
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+      dispatch(getData(path))
+    }, [path]);
+
+    return (
+      <MyComponent 
+        items={items} 
+        status={status} 
+        error={error}
+      />
+    );
+  };
+
+  ConnectedMyComponent.propTypes = {
+    path: PropTypes.string.isRequired,
+  };
+
+  export default ConnectedMyComponent;
+```
+
+```js
+  /** MyComponentSelectors.js*/
+
+  import { MY_SPECIAL_KEY } from './MyComponentConstants';
+  import { selectAPIStatus, selectAPIError, selectAPIResponse } from '../../redux/API/APISelectors';
+
+  // use the same key that you've used in the API action.
+  export const selectItems = state => selectAPIResponse(state, MY_SPECIAL_KEY).items || [];
+
+  export const selectStatus = state => selectAPIStatus(state, MY_SPECIAL_KEY);
+
+  export const selectError = state => selectAPIError(state, MY_SPECIAL_KEY);
```

Comment:
I am wondering If we should encourage consumers to write such a simple selectors.
In this case, won't it make more sense using the api-selectors directly in the `indexjs`?
```js
const status = useSelector(state => selectAPIStatus(state, MY_SPECIAL_KEY));
const error = useSelector(state => selectAPIError(state, MY_SPECIAL_KEY));
```

Reply:
it will make it more usable if other components want to use this selectors, without knowing the `key`